nibabel/xmlbasedimages.py

```diff
@@ -0,0 +1,146 @@
+# emacs: -*- mode: python-mode; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the NiBabel package for the
+#   copyright and license terms.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""
+Thin layer around xml.etree.ElementTree, to abstract nibabel xml support.
+"""
+from xml.etree.ElementTree import Element, SubElement, tostring
+from xml.parsers.expat import ParserCreate
+
+from .filebasedimages import FileBasedHeader, FileBasedImage
+
+
+class XmlSerializable(object):
+    """ Basic interface for serializing an object to xml"""
+
+    def _to_xml_element(self):
+        """ Output should be a xml.etree.ElementTree.Element"""
+        raise NotImplementedError()
+
+    def to_xml(self, enc='utf-8'):
+        """ Output should be an xml string with the given encoding.
+        (default: utf-8)"""
+        return tostring(self._to_xml_element(), enc)
+
+
+class XmlBasedHeader(FileBasedHeader, XmlSerializable):
+    """ Basic wrapper around FileBasedHeader and XmlSerializable."""
+    pass
+
+
+class XmlImageParser(object):
+    """ Base class for defining how to parse xml-based images."""
+
+    HANDLER_NAMES = ['StartElementHandler',
+                     'EndElementHandler',
+                     'CharacterDataHandler']
+
+    def __init__(self, encoding=None, buffer_size=35000000):
+        self.encoding = encoding
+        self.buffer_size = buffer_size
+        self.img = None
+
+    def _create_parser(self):
+        """Internal function that allows subclasses to mess
+        with the underlying parser, if desired."""
+
+        parser = ParserCreate(encoding=self.encoding)  # from xml package
+        parser.buffer_text = True
+        parser.buffer_size = self.buffer_size
```

Comment:
Old code checked before setting, but that's for pre-2.6 compat. Simplify the code...